calicoctl/commands/node/checksystem.go

```diff
@@ -35,16 +35,27 @@ import (
 const minKernelVersion = "2.6.24"
 
 // Required kernel modules to run Calico
-var requiredModules = []string{"ip_set", "ip_tables", "ip6_tables", "ipt_REJECT", "ipt_rpfilter", "ipt_set", "nf_conntrack_netlink",
-	"xt_addrtype", "xt_conntrack", "xt_icmp", "xt_icmp6", "xt_ipvs", "xt_mark", "xt_multiport", "xt_rpfilter", "xt_set", "xt_u32"}
+var requiredModules = map[string]string{"ip_set": "CONFIG_IP_SET", "ip6_tables": "CONFIG_IP6_NF_IPTABLES",
+	"ip_tables": "CONFIG_IP_NF_IPTABLES", "ipt_ipvs": "CONFIG_NETFILTER_XT_MATCH_IPVS", "vfio-pci": "CONFIG_VFIO", "xt_bpf": "CONFIG_BPF", // dataplanes
+	"ipt_REJECT": "CONFIG_NFT_REJECT", "ipt_rpfilter": "CONFIG_IP_NF_MATCH_RPFILTER", "xt_rpfilter": "CONFIG_IP_NF_MATCH_RPFILTER",
+	"ipt_set": "CONFIG_NET_EMATCH_IPSET", "nf_conntrack_netlink": "CONFIG_NF_CT_NETLINK",
+	"xt_addrtype": "CONFIG_NETFILTER_XT_MATCH_ADDRTYPE", "xt_conntrack": "CONFIG_NETFILTER_XT_MATCH_CONNTRACK",
+	"xt_icmp": "icmp", "xt_icmp6": "icmp", "xt_mark": "CONFIG_IP_NF_TARGET_MARK", "xt_multiport": "CONFIG_IP_NF_MATCH_MULTIPORT",
+	"xt_set": "CONFIG_NETFILTER_XT_SET", "xt_u32": "CONFIG_NETFILTER_XT_MATCH_U32"}
+
+// Variable to override bootfile location.
+var overrideBootFile = ""
 
 // Checksystem checks host system for compatible versions
 func Checksystem(args []string) error {
 	doc := `Usage:
-  <BINARY_NAME> node checksystem
+  <BINARY_NAME> node checksystem [--filename=<FILENAME>]
 
 Options:
   -h --help                 Show this screen.
+  -f --filename=<filename>	Override kernel bootfile location, checksystem
```

Comment:
I think this argument isn't super clear to the user - what format is the file supposed to be? Where are the default locations that are checked? etc. 

I think this is referring to the kernel's build configuration file, rather than bootfile? I could be wrong though.

Reply:
Probably worth calling the argument `--kernel-config` or something rather than `--filename` since it's more exact.